WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel+EditProductsM3Tests.swift

```diff
@@ -0,0 +1,173 @@
+import XCTest
+
+@testable import WooCommerce
+@testable import Yosemite
+
+/// The same tests as `DefaultProductFormTableViewModel_EditProductsM2Tests`, but with Edit Products M2 and M3 feature flag on.
+/// When we fully launch Edit Products M2 and M3, we can replace `DefaultProductFormTableViewModel_EditProductsM2Tests` with the test cases here.
+///
+final class DefaultProductFormTableViewModel_EditProductsM3Tests: XCTestCase {
```

Comment:
To answer this question:

> I have copied the approach with DefaultProductFormTableViewModel_EditProductsM2Tests. This seems a bit hard to maintain but I hope that EditProductsM2Tests should be shipped soon?

I think the tests in this class are testing too many things and are not specific enough. Perhaps what we could have done here for the "category" is only test things like 

```swift
fund testViewModelWillShowCategoriesWhenM3FeatureFlagIsOn()
```

And only test this portion:

```swift
            if case .categories(_) = rows[3] {} else {
                XCTFail("Unexpected setting section: \(rows[3])")
            }
```

But no matter, let's leave this as is. 🙂